modules/utility/init.zsh

```diff
@@ -118,13 +118,17 @@ alias ll='ls -lh'        # Lists human readable sizes.
 alias lr='ll -R'         # Lists human readable sizes, recursively.
 alias la='ll -A'         # Lists human readable sizes, hidden files.
 alias lm='la | "$PAGER"' # Lists human readable sizes, hidden files through pager.
-alias lx='ll -XB'        # Lists sorted by extension (GNU only).
 alias lk='ll -Sr'        # Lists sorted by size, largest last.
 alias lt='ll -tr'        # Lists sorted by date, most recent last.
 alias lc='lt -c'         # Lists sorted by date, most recent last, shows change time.
 alias lu='lt -u'         # Lists sorted by date, most recent last, shows access time.
 alias sl='ls'            # I often screw this up.
 
+# Only make alias if ls supports -XB (darwins BSD style ls doesn't)
+if ls -XB &> /dev/null; then
```

Comment:
Wondering if you would consider `if is-callable 'dircolors'` instead.
`ls -XB` might be expensive depending on the directory on which it is executed.

These things keep on adding to the startup time (unless we memoize it on first call).

Reply:
Maybe we could use `ls /dev/null` since that should only be a single file and should be fast.

Reply:
Yes, `ls -XB /dev/null` would be a great idea! That said, would be nice to be consistently using *one* way to detect BSD/GNU versions. Turns out we use `is-callable 'dircolors'` to detect this already.

Reply:
Or just detect the options via `--help` like so:
```shell
if grep -q '\-X' <(ls --help 2>&1); then
  # ...
fi
```
We do similar thing in `rsync` module.

Reply:
@indrajitr I wasn't aware of the performance penalty my idea could introduce, and your concern makes quite a lot of sense.

On the part of finding a better solution:  

At first glance the most straight forward solution @belak has suggested `ls -XB /dev/null` seems feasible, but in the end the `callable 'dircolors'` appears to be the most efficient and elegant alternative. 

But regarding the test with dircolors, I have one question: Is there an unlikeley case I might not see, where dircolors is not callable and gnu-ls is present or the other way around?